browser-switch/build.gradle

```diff
@@ -11,17 +17,55 @@ android {
         minSdkVersion 15
         targetSdkVersion 28
         versionCode 8
-        versionName '0.2.0'
+        versionName version
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    // Ref: https://codelabs.developers.google.com/codelabs/android-room-with-a-view/#2
+    compileOptions {
+        sourceCompatibility = 1.8
+        targetCompatibility = 1.8
+    }
+
+    // robolectric
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+        }
+    }
+    compileOptions {
+        sourceCompatibility = 1.8
+        targetCompatibility = 1.8
     }
 }
 
 dependencies {
-    implementation "androidx.annotation:annotation:${rootProject.ext.androidxVersion}"
-    implementation "androidx.fragment:fragment:${rootProject.ext.androidxVersion}"
+
+    implementation "androidx.appcompat:appcompat:${rootProject.appCompatVersion}"
+    implementation "androidx.annotation:annotation:${rootProject.annotationVersion}"
+    implementation "androidx.fragment:fragment:${rootProject.fragmentVersion}"
 
     testImplementation 'junit:junit:4.12'
-    testImplementation 'org.mockito:mockito-core:2.2.22'
-    testImplementation 'org.robolectric:robolectric:3.3'
+
+    // Ref: https://github.com/powermock/powermock/wiki/Mockito#supported-versions
+    testImplementation 'org.mockito:mockito-core:2.28.2'
+
+    testImplementation 'org.powermock:powermock-module-junit4:2.0.7'
+    testImplementation 'org.powermock:powermock-module-junit4-rule:2.0.7'
+    testImplementation 'org.powermock:powermock-classloading-xstream:2.0.7'
+    testImplementation 'org.powermock:powermock-api-mockito2:2.0.7'
+
+    testImplementation 'org.robolectric:robolectric:4.3'
+
+    def fragment_version = "1.2.4"
+    debugImplementation "androidx.fragment:fragment-testing:${fragment_version}"
+
+    // Ref: https://developer.android.com/training/testing/unit-testing/instrumented-unit-tests
```

Comment:
Do we need all the references in this file? It's nice to link out to references when the code is doing something complicated or confusing, but it doesn't seem like that's necessary here.

Reply:
Yeah technically no. I usually include refs if I feel like a specific piece of code / config will induce a "WTF" and I add the links for closure lol.

Reply:
Yeah, that makes sense. So what's the "WTF" here?

Reply:
Gradle lol. I'll take 'em out.